src/main/java/uk/gov/hmcts/reform/blobrouter/data/DataPersister.java

```diff
@@ -0,0 +1,83 @@
+package uk.gov.hmcts.reform.blobrouter.data;
+
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.reform.blobrouter.data.model.Envelope;
+import uk.gov.hmcts.reform.blobrouter.data.model.NewEnvelope;
+import uk.gov.hmcts.reform.blobrouter.data.model.Status;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+import static java.time.Instant.now;
+
+public class DataPersister {
+
+    private final EnvelopeRepository envelopeRepository;
+    private final EventRecordRepository eventRecordRepository;
+
+    public DataPersister(
+        EnvelopeRepository envelopeRepository,
+        EventRecordRepository eventRecordRepository
+    ) {
+        this.envelopeRepository = envelopeRepository;
+        this.eventRecordRepository = eventRecordRepository;
+    }
+
+    // region BlobProcessor actions
```

Comment:
not sure about those regions, this class shouldn't care who's using its methods imho

Reply:
just thought i'd group for user convenience. regions are nor for class/object per se